### PR TITLE
desktop.space: Fix multioutput rendering

### PR DESCRIPTION
**Pull request by PolyMeilex**

### CODE REVIEW

File: src/desktop/space/mod.rs [hunk 1]
```diff
@@ -600,6 +600,9 @@ impl Space {
                     clear_color,
                     &damage
                         .iter()
+                        // Map from global space to output space
+                        .map(|geo| Rectangle::from_loc_and_size(geo.loc - output_geo.loc, geo.size))
+                        // Map from logical to phisical
```
Comment (vberger) [inline src/desktop/space/mod.rs:605]:
```suggestion
                        // Map from logical to physical
```

File: src/desktop/space/mod.rs [hunk 2]
```diff
@@ -626,6 +629,8 @@ impl Space {
                             .iter()
                             .flat_map(|d| d.intersection(geo))
                             .map(|geo| Rectangle::from_loc_and_size(geo.loc - loc, geo.size))
+                            // Map from global compositor space to output one
+                            .map(|geo| Rectangle::from_loc_and_size(geo.loc - output_geo.loc, geo.size))
```
Comment (Drakulix) [inline src/desktop/space/mod.rs:633]:
```suggestion
                            // Map from global compositor space to output one
                            .map(|geo| Rectangle::from_loc_and_size(geo.loc - output_geo.loc, geo.size))
                            // Map from output space to surface relative coordinates
                            .map(|geo| Rectangle::from_loc_and_size(geo.loc - loc, geo.size))
```

**Review by Drakulix — APPROVED** [review]

File: src/desktop/space/mod.rs (hunk 2 above)
Comment (Drakulix) [reply src/desktop/space/mod.rs:633]: Wait, why is this second one necessary, "loc" already is subtracted by `output_geo.loc` in [line 627](https://github.com/Smithay/smithay/pull/470/files/de0c2e0016498361f40e42b2816c36c6f24f3155#diff-55c984b4634eecc8afa1b426c76e53c642a67f51f7cc73647927edbead721be7R627)?

@PolyMeilex Can you try changing `loc` to only contain `element.location` and nothing else? That should work as well..

Comment (PolyMeilex) [reply src/desktop/space/mod.rs:633]: Yep, that was my initial question on matrix, I'm not sure why it is double substracted.

Comment (PolyMeilex) [reply src/desktop/space/mod.rs:633]: Nope that did not work, it only shows up on the screen when it is subtracted in two places 🤷

Comment (Drakulix) [reply src/desktop/space/mod.rs:633]: Well in that case, I am going to debug that at first, before we merge this. It is just too weird.

**Review by Drakulix — CHANGES_REQUESTED** [review]
See above. Needs more investigation.

**Review by Drakulix — APPROVED** [review]

**Review by vberger — APPROVED** [review]

**Pull request merged**